src/helpers/passable_args/index.js

```diff
@@ -1,32 +1,46 @@
 // @flow
 
+/**
+ * The function which runs the validation tests.
+ *
+ * @callback passableCallback
+ * @param {function} pass
+ * @param {function} enforce
+ */
+
+/**
+ * Get Passable configuration settings
+ * specific - whitelist of tests to run
+ * passes - The function which runs the validations
+ * custom - custom validation rules
+ *
+ * @param {Array.<{specific: String[], passes: passableCallback, custom: Object}>} args - arguments for Passable configuration
+ * @return {object} Passable configuration settings
+ */
 function passableArgs(args: PassableArguments) {
 
     let passes: Function,
-        specific: specific,
-        custom: Rules | void;
-
-    if (args.length === 1) {
-        passes = args[0];
-    } else if (args.length === 3) {
-        specific = args[0];
-        passes = args[1];
-        custom = args[2];
-    } else if (args.length === 2) {
-        if (typeof args[1] === 'function') {
-            specific = args[0];
-            passes = args[1];
-        } else {
-            passes = args[0];
-            custom = args[1];
+        specific: specific = [],
+        custom: Rules = {};
+
+    switch (args.length) {
+      case 0:
+        throw new TypeError('Failed to execute `passableArgs`: At least 1 argument required, but only 0 present.')
+
+      case 1: // [passes] = args;
+        if (typeof args[0] != 'function') {
```

Comment:
There is no need for type coercion. typeof always returns a string. Use `!==`. Same goes for the next case as well.

Reply:
@ealush type coercion in javascript occurs only when comparing operands of different types, such as
```js
'5' == 5;
```
Since in this case, both sides of the comparison are always strings there is no type coercion. and no need to use the `===` operator.

Also, after uglification, the code is transformed to use the `==` operator anyway
```js
typeof args[1] === 'function'  // source code
"function"==typeof e[1]          // dist
```